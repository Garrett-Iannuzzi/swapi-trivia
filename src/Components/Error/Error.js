import React from 'react';
import './Error.scss';
import { Link } from 'react-router-dom';
import images from '../.././images/images.js';


const Error = () => {
  return (
    <section className="section-error">
      <h3 className="header-error">Error!!</h3>
      <p className="p-error">Please fill out all fields before logging in.</p>
      <img className="gif-error" src={images.errorGif} alt="Gif of small bird scared" />
      <Link to='/' className="link-error">
        <button className="button-error">Back To Login</button>
      </Link>
    </section>
  )
}

export default Error;
