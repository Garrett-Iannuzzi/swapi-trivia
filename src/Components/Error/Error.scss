.section-error {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-wrap: nowrap;
  height: 100vh;
  width: 80%;
  margin-top: 10px;
}

.header-error {
  font-size: 3em;
  margin: 0;
  color: indianred;
}

.p-error-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  text-align: center;
  margin: 5px;
  font-size: 1.5em;
  padding: 10px;
  border: indianred 10px solid;
  background-color: black;
  color: white;
  border-radius: 10px;
  opacity: .9;
  width: 350px;
  height: 575px;
}

.p-error {
  margin: 5px;
}

.gif-error {
  height: 350px;
  width: 350px;
  opacity: 1;
  padding: 15px;
  border: indianred 3px solid;
  background-color: indianred;
  border-radius: 10px;
}

.link-error {
  display: flex;
  justify-content: center;
  width: 100%;
  text-decoration: none;
}

.button-error {
  margin: 10px;
  width: 20%;
  font-size: 1em;
  padding: 10px;
  border-radius: 10px;
  background-color: black;
  color: white;
  border: 0;
  cursor: pointer;
}

.button-error:hover {
  background-color: white;
  color: black;
}
