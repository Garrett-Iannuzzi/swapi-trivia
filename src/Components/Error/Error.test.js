import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Error from '../Error/Error.js';

describe('Error', () => {

  it('Should match the snapshot with data passed in', () => {
    const wrapper = shallow(<Error />);

    expect(wrapper).toMatchSnapshot();
  });

})
