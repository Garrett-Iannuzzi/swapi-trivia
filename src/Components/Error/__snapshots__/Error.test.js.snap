// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Should match the snapshot with data passed in 1`] = `
<section
  className="section-error"
>
  <div
    className="p-error-container"
  >
    <h3
      className="header-error"
    >
      Error!!
    </h3>
    <p
      className="p-error"
    >
      *All fields must be filled out to login*
    </p>
    <p
      className="p-error"
    >
      *Please return to login*
    </p>
    <img
      alt="Gif of small bird scared"
      className="gif-error"
      src="errorGif.gif"
    />
  </div>
  <Link
    className="link-error"
    to="/"
  >
    <button
      className="button-error"
    >
      Back To Login
    </button>
  </Link>
</section>
`;
